### PR TITLE
client/web: hook up data fetching to fill --dev React UI

### PR DESCRIPTION
**Pull request by soniaappasamy**
This hooks up the GET endpoint. POST to come in a follow-up.

Updates tailscale/corp#13775

### CODE REVIEW

File: client/web/src/hooks/node-data.ts [hunk 1]
```diff
@@ -20,6 +16,12 @@ export type NodeData = {
...
 
+export type UserProfile = {
+  LoginName: string
+  DisplayName: string
+  ProfilePicURL: string
+}
+
 // testData is static set of nodedata used during development.
 // This can be removed once we have a real node data API.
 const testData: NodeData = {
```
Comment (willnorris) [inline client/web/src/hooks/node-data.ts:27]: I think we can probably remove the testData?

File: client/web/web.go [hunk 2]
```diff
@@ -411,6 +374,16 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		data.IP = st.TailscaleIPs[0].String()
 	}
 
+	if asJSON {
+		if err := json.NewEncoder(w).Encode(data); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
```
Comment (willnorris) [inline client/web/web.go:383]: this is getting logged in output:

```
superfluous response.WriteHeader call from tailscale.com/client/web.(*Server).serveGetNodeData (web.go:383)
```

I think at this point, a 200 response header has already been written

File: client/web/web.go [hunk 3]
```diff
@@ -309,80 +300,52 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+	TUNMode           bool
+	IsSynology        bool
+	DSMVersion        int // 6 or 7, if IsSynology=true
+	IsUnraid          bool
+	UnraidToken       string
+	IPNVersion        string
+}
 
-	st, err := s.lc.Status(ctx)
+func (s *Server) serveGetNodeData(w http.ResponseWriter, r *http.Request, user string, asJSON bool) {
```
Comment (willnorris) [inline client/web/web.go:333]: I think overloading this method to handle both the old static index page, as well as the JSON raw data output for the new page is somewhat confusing.  

Maybe there is a single `getNodeData()` method that returns a `nodeData` struct.  And then separate handler methods for the old go-template index page and the new `/api/data` JSON endpoint, each of which call `getNodeData()` ?

**Review by willnorris — APPROVED** [review]
a couple of inline comments, otherwise lgtm

File: client/web/src/hooks/node-data.ts (hunk 1 above)
Comment (soniaappasamy) [reply client/web/src/hooks/node-data.ts:27]: whoop! yes!

File: client/web/web.go (hunk 3 above)
Comment (soniaappasamy) [reply client/web/web.go:333]: Figured we'd be getting rid of it once we cut out the old html templates, but I like your suggestion - a little cleaner in the interim.

**Pull request merged**